fortran/Makefile

```diff
@@ -1,66 +1,75 @@
 FC := gfortran
+FCNAME := $(notdir $(FC))
 
-LIBNAME   := libncclfor.so
-LIBSONAME := $(patsubst %,%.$(NCCL_MAJOR),$(LIBNAME))
-LIBTARGET := $(patsubst %,%.$(NCCL_MAJOR).$(NCCL_MINOR).$(NCCL_PATCH),$(LIBNAME))
-LIBLINK   += $(patsubst lib%.so,-l%,$(LIBNAME))
+BUILDDIR ?= ../build
+INCDIR := $(BUILDDIR)/include
+LIBDIR := $(BUILDDIR)/lib
+OBJDIR := $(BUILDDIR)/obj
 
-ifeq ($(FC), gfortran)
-FCMODFLAGS  := -J$(INCDIR)
-FCPREFLAGS  := -cpp
-FCCUDAFLAGS :=
-FCFLAGS     := -ffree-line-length-none
-CUDAFORDEP  := libcudafor
-CUDALINK    := -L$(CUDA_LIB) -lcudart
-CUDAFORLINK := -lcudafor
-else ifeq ($(FC), ifort)
-FCMODFLAGS  := -module $(INCDIR)
-FCPREFLAGS  := -fpp
-FCCUDAFLAGS :=
-FCFLAGS     :=
-CUDAFORDEP  := libcudafor
-CUDALINK    := -L$(CUDA_LIB) -lcudart
-CUDAFORLINK := -lcudafor
-else
+LIBNAME    := libncclfor.so
+LIBSONAME  := $(patsubst %,%.$(NCCL_MAJOR),$(LIBNAME))
+LIBTARGET  := $(patsubst %,%.$(NCCL_MAJOR).$(NCCL_MINOR).$(NCCL_PATCH),$(LIBNAME))
+LIBLINK    += $(patsubst lib%.so,-l%,$(LIBNAME))
+
+LIBCUDAFOR := libcudafor.so
+
+ifeq ($(FCNAME), pgf90)
```

Comment:
Ok, I see why you did that now. Could we just compare against pgf* then ? The former way of testing if it is not gfortran and not ifort then it is pgfortran makes the whole thing confusing.